sources/packages/api/src/route-controllers/application/application.controller.ts

```diff
@@ -66,8 +68,17 @@ export class ApplicationController extends BaseController {
     @Body() payload: CreateApplicationDto,
     @UserToken() userToken: IUserToken,
   ): Promise<number> {
+    const programYear = await this.programYearService.getProgramYear(
+      payload.programYearId,
+    );
+
+    if (!programYear) {
+      throw new BadRequestException(
```

Comment:
IMO option would be more a 422 Unprocessable Entity exception.
I would say that we should use Bad Request when the request is not well-formed, like the ones that nest throws when "class-validators" fails. In this case, the request was understood, but the data could not be processed.

Reply:
I actually did as per the requirement, but fair enough

 Validates if the program year is active, if not throw a BadRequestException or similar.